### PR TITLE
[PHP 8.4] Add `ext-intl` constant types

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Part of #3872

### CODE REVIEW

File: reference/intl/normalizer.xml [hunk 1]
```diff
@@ -175,6 +175,12 @@
       </row>
      </thead>
      <tbody>
+      <row>
+       <entry>8.4.0</entry>
+       <entry>
+        The class constants are now typed.
+       </entry>
+      </row>
```
Comment (Girgias) [inline reference/intl/normalizer.xml:183]: Were the class constant types already documented?

File: reference/intl/numberformatter-constants.xml [hunk 2]
```diff
@@ -722,11 +793,13 @@
   <para>
    Pad position values used by <function>numfmt_get_attribute</function>
    and <function>numfmt_set_attribute</function> with
-   <constant>NumberFormatter::PADDING_POSITION</constant> attribute.
+   <constant>NumberFormatter::PADDING_POSITION</constant>
+      <type>int</type> attribute.
```
Comment (Girgias) [inline reference/intl/numberformatter-constants.xml:797]: This is incorrect

File: reference/intl/numberformatter-constants.xml [hunk 3]
```diff
@@ -639,11 +702,13 @@
   <para>
    Rounding mode values used by <function>numfmt_get_attribute</function>
    and <function>numfmt_set_attribute</function> with
-   <constant>NumberFormatter::ROUNDING_MODE</constant> attribute.
+   <constant>NumberFormatter::ROUNDING_MODE</constant>
+      <type>int</type> attribute.
```
Comment (Girgias) [inline reference/intl/numberformatter-constants.xml:706]: Ditto

File: reference/intl/dateformatter.xml [hunk 4]
```diff
@@ -128,6 +128,28 @@
...
+       <entry>&Description;</entry>
+      </row>
+     </thead>
+     <tbody>
+      <row>
+       <entry>8.4.0</entry>
+       <entry>
+        The class constants are now typed.
+       </entry>
+      </row>
```
Comment (Girgias) [inline reference/intl/dateformatter.xml:148]: Is type already documented?

File: reference/intl/normalizer.xml (hunk 1 above)
Comment (alexandre-daubois) [reply reference/intl/normalizer.xml:183]: They were already documented but without the types. However, the classes stubs are already up to date so everything's fine on this side

File: reference/intl/intlchar.xml [hunk 5]
```diff
@@ -4019,28 +4019,40 @@
     &reftitle.constants;
     <variablelist>
      <varlistentry xml:id="intlchar.constants.unicode-version">
-      <term><constant>IntlChar::UNICODE_VERSION</constant></term>
+      <term>
+      <constant>IntlChar::UNICODE_VERSION</constant>
+      <type>string</type>
+     </term>
```
Comment (Girgias) [inline reference/intl/intlchar.xml:4025]: Indent is wrong

File: reference/intl/numberformatter-constants.xml (hunk 2 above)
Comment (alexandre-daubois) [reply reference/intl/numberformatter-constants.xml]: I went a bit too far with the regex replacement 😄 It's updated now

**Review by Girgias — APPROVED** [review]

**Pull request merged**